### PR TITLE
fix: Prioritise Vercel/Netlify env settings over the default URLs

### PR DESCRIPTION
**Pull request by AdrianLThomas**
**_What will change?_**

Tweaked the configuration files so that if the `NEXT_PUBLIC_BASE_URL` is already set then it is used in place of the hosting provider URL.

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -1,2 +1,2 @@
 [build]
-  command = "NEXT_PUBLIC_BASE_URL=$URL yarn build"
+  command = "NEXT_PUBLIC_BASE_URL=\"${URL:-$URL}\" yarn build"
```
Comment (mattvanvoorst-contentful) [inline netlify.toml:2]: Should the first `URL` be `NEXT_PUBLIC_BASE_URL `

Comment (AdrianLThomas) [reply netlify.toml:2]: no

Comment (AdrianLThomas) [reply netlify.toml:2]: maybe

Comment (AdrianLThomas) [reply netlify.toml:2]: will fix, thanks

**Review by mattvanvoorst-contentful — APPROVED** [review]

**Review by cf-remylenoir — APPROVED** [review]

**Pull request merged**